File: backend/internal/models/argocd.go
package models

import "time"

type ArgoApplication struct {
	ID             string    `json:"id"`
	UserID         string    `json:"user_id"`
	Path           string    `json:"path"`
	Name           string    `json:"name"`
	RepoURL        string    `json:"repo_url"`
	TargetRevision string    `json:"target_revision"`
	Destination    string    `json:"destination"` // Server URL
	Content        string    `json:"content"`
	CreatedAt      time.Time `json:"created_at"`
	UpdatedAt      time.Time `json:"updated_at"`
}

// ArgoAppDetail represents the runtime state of an application
type ArgoAppDetail struct {
	Name      string           `json:"name"`
	Project   string           `json:"project"`
	Source    ArgoSource       `json:"source"`
	Dest      ArgoDestination  `json:"destination"`
	Sync      ArgoSyncStatus   `json:"sync"`
	Health    ArgoHealthStatus `json:"health"`
	Operation *ArgoOperation   `json:"operation,omitempty"`
}

type ArgoSource struct {
	RepoURL        string `json:"repoURL"`
	Path           string `json:"path"`
	TargetRevision string `json:"targetRevision"`
}

type ArgoDestination struct {
	Server    string `json:"server"`
	Namespace string `json:"namespace"`
}

type ArgoSyncStatus struct {
	Status   string `json:"status"`   // Synced, OutOfSync
	Revision string `json:"revision"` // Git revision
}

type ArgoHealthStatus struct {
	Status  string `json:"status"`  // Healthy, Degraded, Progressing, Missing, Suspended, Unknown
	Message string `json:"message"` // Detailed message
}

type ArgoOperation struct {
	Sync *ArgoSyncOperation `json:"sync"`
}

type ArgoSyncOperation struct {
	Revision string `json:"revision"`
	Status   string `json:"status"`
}

File: frontend/lib/models/argocd.dart
class ArgoAppDetail {
  final String name;
  final String project;
  final ArgoSource source;
  final ArgoDestination destination;
  final ArgoSyncStatus sync;
  final ArgoHealthStatus health;
  final ArgoOperation? operation;

  ArgoAppDetail({
    required this.name,
    required this.project,
    required this.source,
    required this.destination,
    required this.sync,
    required this.health,
    this.operation,
  });

  factory ArgoAppDetail.fromJson(Map<String, dynamic> json) {
    return ArgoAppDetail(
      name: json['name'] ?? '',
      project: json['project'] ?? '',
      source: ArgoSource.fromJson(json['source'] ?? {}),
      destination: ArgoDestination.fromJson(json['destination'] ?? {}),
      sync: ArgoSyncStatus.fromJson(json['sync'] ?? {}),
      health: ArgoHealthStatus.fromJson(json['health'] ?? {}),
      operation: json['operation'] != null ? ArgoOperation.fromJson(json['operation']) : null,
    );
  }
}

class ArgoSource {
  final String repoURL;
  final String path;
  final String targetRevision;

  ArgoSource({
    required this.repoURL,
    required this.path,
    required this.targetRevision,
  });

  factory ArgoSource.fromJson(Map<String, dynamic> json) {
    return ArgoSource(
      repoURL: json['repoURL'] ?? '',
      path: json['path'] ?? '',
      targetRevision: json['targetRevision'] ?? '',
    );
  }
}

class ArgoDestination {
  final String server;
  final String namespace;

  ArgoDestination({
    required this.server,
    required this.namespace,
  });

  factory ArgoDestination.fromJson(Map<String, dynamic> json) {
    return ArgoDestination(
      server: json['server'] ?? '',
      namespace: json['namespace'] ?? '',
    );
  }
}

class ArgoSyncStatus {
  final String status;
  final String revision;

  ArgoSyncStatus({
    required this.status,
    required this.revision,
  });

  factory ArgoSyncStatus.fromJson(Map<String, dynamic> json) {
    return ArgoSyncStatus(
      status: json['status'] ?? 'Unknown',
      revision: json['revision'] ?? '',
    );
  }
}

class ArgoHealthStatus {
  final String status;
  final String message;

  ArgoHealthStatus({
    required this.status,
    required this.message,
  });

  factory ArgoHealthStatus.fromJson(Map<String, dynamic> json) {
    return ArgoHealthStatus(
      status: json['status'] ?? 'Unknown',
      message: json['message'] ?? '',
    );
  }
}

class ArgoOperation {
  final ArgoSyncOperation? sync;

  ArgoOperation({this.sync});

  factory ArgoOperation.fromJson(Map<String, dynamic> json) {
    return ArgoOperation(
      sync: json['sync'] != null ? ArgoSyncOperation.fromJson(json['sync']) : null,
    );
  }
}

class ArgoSyncOperation {
  final String revision;
  final String status;

  ArgoSyncOperation({
    required this.revision,
    required this.status,
  });

  factory ArgoSyncOperation.fromJson(Map<String, dynamic> json) {
    return ArgoSyncOperation(
      revision: json['revision'] ?? '',
      status: json['status'] ?? '',
    );
  }
}
